src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/span_links/index.tsx

```diff
@@ -0,0 +1,287 @@
+    ({ rowIndex, columnId }) => {
+      const item = filteredItems[rowIndex];
+      switch (columnId) {
+        case 'span':
+          return (
+            <EuiTextTruncate
+              data-test-subj={`${type}-spanName-${item.spanId}`}
+              text={item.details?.spanName || 'N/A'}
+            />
+          );
+        case 'duration':
+          return <Duration duration={item.details?.duration || 0} />;
+        case 'serviceName':
+          return (
+            <ServiceNameWithIcon
+              agentName={item.details?.agentName}
+              serviceName={item.details?.serviceName || 'N/A'}
+              truncate
+            />
+          );
+        case 'traceId':
+          return <EuiTextTruncate text={item.traceId} />;
+      }
+    },
+    [filteredItems, type]
+  );
+
+  useEffect(() => {
+    if (type === 'incoming' && value?.incomingSpanLinks.length === 0) {
+      setType('outgoing');
+    }
+  }, [value, type]);
+
+  const selectOptions: EuiSelectOption[] = useMemo(
+    () => [
+      {
+        'data-test-subj': 'spanLinkTypeSelect-incoming',
+        value: 'incoming',
+        text: i18n.translate(
+          'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.combo.incomingLinks',
+          {
+            defaultMessage: 'Incoming links ({incomingLinks})',
+            values: { incomingLinks: value?.incomingSpanLinks.length },
+          }
+        ),
+        disabled: !value?.incomingSpanLinks.length,
+      },
+      {
+        'data-test-subj': 'spanLinkTypeSelect-outgoing',
+        value: 'outgoing',
+        text: i18n.translate(
+          'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.combo.outgoingLinks',
+          {
+            defaultMessage: 'Outgoing links ({outgoingLinks})',
+            values: { outgoingLinks: value?.outgoingSpanLinks.length },
+          }
+        ),
+        disabled: !value?.outgoingSpanLinks.length,
+      },
+    ],
+    [value]
+  );
+
+  if (
+    loading ||
+    (!error && value?.incomingSpanLinks.length === 0 && value?.outgoingSpanLinks.length === 0)
+  ) {
+    return null;
+  }
+
+  return (
+    <EuiAccordion
+      data-test-subj="spanLinksAccordion"
+      id="spanLinksAccordion"
+      buttonContent={
+        <strong>
+          {i18n.translate('unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks', {
+            defaultMessage: 'Span Links',
+          })}
+        </strong>
+      }
+      initialIsOpen
+      paddingSize="none"
+    >
+      <EuiSpacer size="s" />
+      {error ? (
+        <EuiText color="subdued">
+          {i18n.translate('unifiedDocViewer.observability.traces.docViewerSpanOverview.error', {
+            defaultMessage: 'An error happened when trying to fetch data. Please try again',
+          })}
+        </EuiText>
+      ) : null}
+
+      <EuiFlexGroup direction="column" gutterSize="s">
```

Comment:
When an error occurs, the content renders with error and the empty message. Was this the intended behavior?